framework/include/kernels/ADCoupledTimeDerivative.h

```diff
@@ -0,0 +1,38 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef ADCOUPLEDTIMEDERIVATIVE_H
+#define ADCOUPLEDTIMEDERIVATIVE_H
+
+#include "ADKernel.h"
+
+// Forward Declaration
+template <ComputeStage>
+class ADCoupledTimeDerivative;
+
+declareADValidParams(ADCoupledTimeDerivative);
+
+/**
+ * This calculates the time derivative for a coupled variable
+ **/
+template <ComputeStage compute_stage>
+class ADCoupledTimeDerivative : public ADKernel<compute_stage>
```

Comment:
Any reason not to do `ADKernelValue` here?

Reply:
Right, I keep forgetting about that!

Reply:
done. Also used ADKernelGrad in `ADSplitCHWResBase`